metrics-datadog/src/main/java/org/coursera/metrics/datadog/DynamicTagsCallback.java

```diff
@@ -9,9 +9,12 @@
  */
 public interface DynamicTagsCallback {
   /**
-   *
-   * @return dynamic tags that will merge into the static tags. Dynamic tags will overwrite
-   * static tags with the same key
+   * Allow generating tags from the metric name dynamically.
+   * @param metricName is the metric name to generate dynamic tags from (if desired)
+   * @param expansion this is a string that gets appended (joined with a period) to form the full
+   *                  metric name.
+   * @return dynamic tags that will merge into the static tags
    */
-  public List<String> getTags();
```

Comment:
No reason to break backwards compatibility here - we can leave this in and our implementation returns an empty list

Reply:
Oh actually, the call to getTags() is replaced by generateTagsFromMetricName().  The two pretty much occupy the same space.  We could add logic to determine calling one vs the other (like a flag in the constructor), but it is probably just overall easier to call the new one with null args.

I could call the new one getTags(arg1, arg2) instead of  generateTagsFromMetricName(arg1, arg2)?  That might make it easier to swallow?

Reply:
I'm fine with it, ship it!